cmake/TestUtils.cmake

```diff
@@ -18,6 +18,11 @@ function(build_test SRCFILE LINK_LIBRARIES PREPROC_DEFS)
   if (TARGET gtest)
     add_dependencies(${target} gtest) # make sure gtest is built first
   endif()
+  # openmp required by some tests using for instance PowerSpectrum: provided by iomp if MKL ON
+  if (NOT FL_LIBRARIES_USE_MKL)
+    list(APPEND LINK_LIBRARIES OpenMP::OpenMP_CXX)
```

Comment:
Using a global property is definitely not what we want here.

If anything, we'd want to use `target_link_libraries`, but this shouldn't be necessary here should we should be able to include these linker flags from a link call to the target that requires OpenMP in the first place.

Reply:
sure, I tried to just target_link_libraries(OpenMP_CXX) but then nvcc was failing because not aware of the openmp flag. Retrying...

Reply:
done